packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -582,7 +585,7 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     postBody += `${key}: ${value}\r\n`;
                 }
             });
-            const sharingLink = await this.sharingLinkP;
+            const sharingLink = await this.appSharingLinkP;
```

Comment:
I'd love for us to get rid of this await to avoid the perf issue from coming back later

Reply:
1.) I think we can use the name sharingLinkToRedeem/redeemSharingLink for this.
2.) Resolved url is the only way to communicate stuff from resolver to driver storage.
3.) So we can just set sharinglink in case it is redeem sharing link and use the cache(within resolver) for sharing link generated by resolver just like before. By this way we can avoid storing it as promise.

Reply:
@leeviana Do we need to store resolver generated sharing link on the resolved url for any purpose?

Reply:
I think the promise is fine, the issue is actually calling await on it here, right? The resolver just needs it for getAbsoluteUrl and can get it from the cache or the iresolvedurl data